framework/test/mocha/unit/modules/chain/submodules/process_transactions.js

```diff
@@ -41,16 +80,435 @@ describe('ProcessTransactions', () => {
 		},
 	};
 
-	beforeEach(async () => {
+	beforeEach(done => {
 		// Act
-		processTransactions = new ProcessTransactions(() => {}, {
-			components: {},
-			logic: {},
+		processTransactions = new ProcessTransactions(() => {
+			processTransactions.onBind(paramScope);
+
+			// Add matcher to transactions
+			trs1.matcher = () => true;
+			trs2.matcher = () => true;
+
+			// Add prepare steps to transactions
+			trs1.prepare = sinonSandbox.stub();
```

Comment:
prefer stubbing with `stub.trs1Prepare = sinonSandbox.stub(trs1, 'prepare')`. Same goes with other stubbings. 

This way, `sinonSandbox.restore()` could work. At the moment, sinon is not able to track the stubs, and the reason tests are working is we are reassigning new stubs to these methods in every beforeEach hook.

Then instead of 
```js
			trs1.validate = sinonSandbox.stub().returns(validResponse);
```

we would use 
```js
			stubs.trs1Validate.returns(validResponse);
```

Reply:
@yatki I love using `sinonSandbox.stub(trs1, 'prepare')` this convention, but problem in this case is we don't have `prepare`, `apply` on those trs objets. So I have to assign those.